### ISSUE TITLE
Are there any plans to move to Antlr4?

### ISSUE THREAD

**Issue by parrotcar00** [eclipse/xtext#1929]
I see that Xtext uses Antl3 grammar under the hood. Are there any plans to start using Antlr4?

**Comment by cdietrich** [comment]
no. there are no plans

**Comment by miklossy** [comment]
See also https://www.eclipse.org/forums/index.php?t=msg&th=1084702&goto=1767965&#msg_1767965

**Comment by parrotcar00** [comment]
@miklossy Thanks!
I understand that Antlr4 has undergone bunch of changes in the backend and its API, but are there any changes in the syntax of the grammar files themselves?

I'm planning on doing a desing of DSL in Antlr4 but I think I should be fine using Xtext as the grammar syntax should be mostly same across Antlr 3 and 4, right?

**Comment by szarnekow** [comment]
Joining the discussion here to clarify a few things:
Antlr3 offers very powerful static analysis features (if backtracking is disabled). They help users to figure if their language is ambiguous or not. To my best knowledge, Antlr4 resorts to runtime-disambiguation which is only good idea at the first glance. We also did some careful tuning to the way the Xtext grammar is translated to Antlr3. *All* performance analysis that we did with Antlr4 showed that it is a little slower even when all known best practices are applied. Not dramatically slower but a little (also related to the way the predication works). Both of these conclusions don't help to make a compelling argument to exchange the backend with Antlr4 in Xtext.

About the Xtext grammar language itself: Even *if* there was a rewrite towards Antlr4 as the parser technology, the grammar language itself would remain fully backwards compatible. As a user, you wouldn't notice a difference.
